### PR TITLE
Enables selecting and unselecting kernel tracepoints in the UI and rec…

### PR DESCRIPTION
**Pull request by MariaSandru27**
…ords which ones have been chosen

### CODE REVIEW

**Review by danielfenner — COMMENTED** [review]
looks good, merely nits ...

File: OrbitGl/DataManager.cpp [hunk 1]
```diff
@@ -141,3 +141,17 @@ const std::shared_ptr<Process>& DataManager::selected_process() const {
   CHECK(std::this_thread::get_id() == main_thread_id_);
   return selected_process_;
 }
+
+void DataManager::SelectTracepoint(const TracepointInfo& info) {
+  std::pair<std::string, std::string> pair = std::make_pair(info.category(), info.name());
+  if (!selected_tracepoints_.contains(pair)) selected_tracepoints_.emplace(pair);
```
Comment (dimitry-) [inline OrbitGl/DataManager.cpp:147]: nitpick: you can actually do `emplace(info.category(), info.name());` it should create the pair in place.

**Review by dimitry- — APPROVED** [review]
otherwise lgtm.

File: OrbitGl/TracepointsDataView.cpp [hunk 2]
```diff
@@ -6,6 +6,12 @@
 
 #include "App.h"
 
+namespace {
+static const std::string kMenuActionSelect = "Hook";
+;
```
Comment (danielfenner) [inline OrbitGl/TracepointsDataView.cpp:11]: Remove the line with only the `;`

File: OrbitGl/DataManager.h [hunk 3]
```diff
@@ -13,11 +13,28 @@
...
+    return std::hash<std::string>{}(info.category()) * 37 + std::hash<std::string>{}(info.name());
+  }
+};
+
+struct EqualTracepointInfo {
+  size_t operator()(const TracepointInfo& left, const TracepointInfo& right) const {
+    return left.category().compare(right.category()) == 0;
+  }
+};
+}  // namespace internal
```
Comment (danielfenner) [inline OrbitGl/DataManager.h:37]: Add empty line after` }  // namespace internal`

File: OrbitGl/TracepointsDataView.cpp [hunk 4]
```diff
@@ -89,10 +97,35 @@ std::vector<std::string> TracepointsDataView::GetContextMenu(
...
+      GOrbitApp->SelectTracepoint(GetTracepoint(i));
+    }
+  } else if (action == kMenuActionUnselect) {
+    for (int i : item_indices) {
+      GOrbitApp->DeselectTracepoint(GetTracepoint(i));
+    }
+  } else {
+    DataView::OnContextMenu(action, menu_index, item_indices);
+  }
+}
```
Comment (danielfenner) [inline OrbitGl/TracepointsDataView.cpp:128]: Add line between the two functions.

**Review by danielfenner — APPROVED** [review]
Merly nits. Please fix and merge... LGTM!

File: OrbitGl/DataManager.h [hunk 5]
```diff
@@ -13,11 +13,29 @@
...
+namespace internal {
+struct HashTracepointInfo {
+  size_t operator()(const TracepointInfo& info) const {
+    return std::hash<std::string>{}(info.category()) * 37 + std::hash<std::string>{}(info.name());
+  }
+};
+
+struct EqualTracepointInfo {
+  size_t operator()(const TracepointInfo& left, const TracepointInfo& right) const {
+    return left.category().compare(right.category()) == 0;
```
Comment (dimitry-) [inline OrbitGl/DataManager.h:34]: I think you forgot to compare the name?

Comment (dimitry-) [reply OrbitGl/DataManager.h:34]: and equal returns bool not size_t.

Comment (danielfenner) [reply OrbitGl/DataManager.h:34]: +1 , should be something like
`left.category() == right.category() && left.name() == right.name()`

**Pull request merged**